src/cms/models/pages/imprint_page_translation.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+from django.conf import settings
+from django.db import models
+
+from .abstract_base_page_translation import AbstractBasePageTranslation
+
+from .imprint_page import ImprintPage
+from ..languages.language import Language
+
+
+logger = logging.getLogger(__name__)
+
+
+class ImprintPageTranslation(AbstractBasePageTranslation):
+    """
+    Data model representing a imprint translation
+
+    :param id: The database id of the imprint translation
+    :param slug: The slug identifier of the translation (unique per :class:`~cms.models.regions.region.Region` and
+                 :class:`~cms.models.languages.language.Language`)
+    :param status: The status of the imprint translation (choices: :mod:`cms.constants.status`)
+    :param title: The title of the imprint translation
+    :param text: The content of the imprint translation
+    :param currently_in_translation: Flag to indicate a translation is being updated by an external translator
+    :param version: The revision number of the imprint translation
+    :param minor_edit: Flag to indicate whether the difference to the previous revision requires an update in other
+                       languages
+    :param created_date: The date and time when the imprint translation was created
+    :param last_updated: The date and time when the imprint translation was last updated
+
+    Relationship fields:
+
+    :param page: The imprint pag
+    e the translation belongs to (related name: ``translations``)
+    :param language: The language of the imprint translation (related name: ``imprint_translations``)
+    :param creator: The user who created the imprint translation (related name: ``imprint_translations``)
+    """
+
+    page = models.ForeignKey(
+        ImprintPage, related_name="translations", on_delete=models.CASCADE
+    )
+    language = models.ForeignKey(
+        Language, related_name="imprint_translations", on_delete=models.CASCADE
+    )
+    creator = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="imprint_translations",
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+
+    @property
+    def permalink(self):
+        """
+        This property calculates the permalink dynamically by joining the parent path together with the slug
+
+        :return: The permalink of the imprint
+        :rtype: str
+        """
+        return "/".join(
+            filter(
+                None,
+                [
+                    self.page.region.slug,
+                    self.language.code,
+                    "disclaimer",
```

Comment:
Should this be hardcoded here?

Reply:
Probably not and also probably related to the question regarding tree related functions below: https://github.com/Integreat/cms-django/pull/488#discussion_r491673129